engine/src/cmd/damageable.h

```diff
@@ -26,78 +26,142 @@
+ */
+class Damageable : public DamageableObject
 {
 public:
-    // TODO: Consider making into protected later
-    // Fields
-    // Inconsistency between this (0) and Init (1) for hull and maxhull.
-    // Chose Init and value of 1. Also commented out was 10.
-    // Doesn't really matter, but documenting anyway.
-    float maxhull = 1;
-    float hull = 1;
-    Armor armor;
-    Shield shield;
+    DamageableLayer *hull;
+    DamageableLayer *armor;
+    DamageableLayer *shield;
+
+    float *current_hull;
+    float *max_hull;
+
 
+    bool dying;
+    //Is dead already?
+    bool  killed;
 
     // Methods
 public:
-  Damageable();
+    Damageable(): hull(&layers[0]),
+        armor(&layers[1]),
+        shield(&layers[2]),
+        current_hull(&hull->facets[0].health),
+        max_hull(&hull->facets[0].factory_max_health),
+        dying(false),
+        killed(false) {}
 
 protected:
-  virtual ~Damageable() = default;
-  // forbidden
-  Damageable( const Damageable& ) = delete;
-  // forbidden
-  Damageable& operator=( const Damageable& ) = delete;
+    virtual ~Damageable() = default;
+    // forbidden
+    Damageable( const Damageable& ) = delete;
+    // forbidden
+    Damageable& operator=( const Damageable& ) = delete;
 
 public:
-  float GetHull() const { return hull;}
-
-  void ArmorData( float armor[8] ) const;
-
-  // Is the shield up from this position
-  bool ShieldUp( const Vector& ) const;
-
-  //applies damage from the given pnt to the shield, and returns % damage applied and applies lighitn
-  virtual float DealDamageToShield( const Vector &pnt, float &Damage );
-
-  float FShieldData() const;
-  float RShieldData() const;
-  float LShieldData() const;
-  float BShieldData() const;
-//short fix
-//  void ArmorData( float armor[8] ) const;
-//Gets the current status of the hull
-//  float GetHull() const
-//  {
-//      return hull;
-//  }
-  float GetHullPercent() const
-  {
-      return maxhull != 0 ? hull/maxhull : hull;
-  }
-
-  //reduces shields to X percentage and reduces shield recharge to Y percentage
-  void leach( float XshieldPercent, float YrechargePercent, float ZenergyPercent );
-
-  virtual float DealDamageToHull( const Vector &pnt, float Damage );
-
-  float MaxShieldVal() const;
-//regenerates all 2,4, or 6 shields for 1 SIMULATION_ATOM
-  virtual void RegenShields() = 0;
-  bool withinShield( const ShieldFacing &facing, float theta, float rho );
-
-  static float totalShieldVal( const Shield &shield );
-  static float currentTotalShieldVal( const Shield &shield );
-  static float totalShieldEnergyCapacitance( const Shield &shield );
+    // We follow the existing convention of GetX for the actual health value
+    // because we are constrained by existing python interfaces, which cannot
+    // be easily changed.
+    const float GetHull() const {
+        return hull->facets[0].health;
+    }
+
+    const float GetArmor(int facet = 0) const {
+        return armor->facets[facet].health;
```

Comment:
what are the valid values of `facets`? Using an enum would convey that and help limit to valid values.

Reply:
GetArmor and GetShield are not used. I added them for completeness sake because we have GetHull and it is exposed to python. We can probably remove this but if we expose hull to python, we should also expose these. Because it's python, you can't limit to enum class anyway (can you?!?). I added a TODO to check for ranges.

Reply:
You should still be able to limit to the enum class too, it's just a matter of mapping; it might be easier to do in the Python C API as opposed to Boost::Python though...not sure. But if it's too complex we can come back to it in a later refactor.  I'm mostly looking to make sure we can understand the code; for now, comments might be best so we an convey intent.